pandas/plotting/_matplotlib/tools.py

```diff
@@ -375,7 +392,7 @@ def _get_all_lines(ax):
     return lines
 
 
-def _get_xlim(lines) -> Tuple[float, float]:
+def _get_xlim(lines: List["Line2D"]) -> Tuple[float, float]:
```

Comment:
Iterable,

from https://github.com/python/typeshed/blob/master/CONTRIBUTING.md

> Some further tips for good type hints:

> avoid invariant collection types (List, Dict) in argument positions, in favor of covariant types like Mapping or Sequence;

Reply:
isnt more specific better when its accurate?

Reply:
for return types yes, but for inputs should be as permissible as possible, so in this function the use of lines is just `for l in lines`, so iterable can be safely passed.

Reply:
> but for inputs should be as permissible as possible

isnt that principle primarily for user-facing functions?

Will update.

Reply:
I think that is also good internally as in general facilitates reusability, can avoid unnecessary type conversions, but most importantly prevents us accidentally mutating a passed list.